handlers/bounty.go

```diff
@@ -373,3 +374,163 @@ func MakeBountyPayment(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 }
+
+func BountyBudgetWithdraw(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	pubKeyFromAuth, _ := ctx.Value(auth.ContextKey).(string)
+
+	if pubKeyFromAuth == "" {
+		fmt.Println("no pubkey from auth")
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	request := db.WithdrawBudgetRequest{}
+	body, err := io.ReadAll(r.Body)
+	r.Body.Close()
+
+	err = json.Unmarshal(body, &request)
+	if err != nil {
+		w.WriteHeader(http.StatusNotAcceptable)
+		return
+	}
+
+	decodedInvoice, err := decodepay.Decodepay(request.PaymentRequest)
+	amountInt := decodedInvoice.MSatoshi / 1000
+	amount := uint(amountInt)
+
+	if err == nil {
+		// check if the orgnization bounty balance
+		// is greater than the amount
+		orgBudget := db.DB.GetOrganizationBudget(request.OrgUuid)
```

Comment:
do we ever check if that user is the owner of the org? what if I pass my pubkey but your orguuid?

Reply:
I missed this, I have pushed an update.